server/balancer_test.go

```diff
@@ -743,3 +763,97 @@ func checkTransferLeader(c *C, bop Operator, sourceID, targetID uint64) {
 	c.Assert(op.OldLeader.GetStoreId(), Equals, sourceID)
 	c.Assert(op.NewLeader.GetStoreId(), Equals, targetID)
 }
+
+var _ = Suite(&testBalanceHotRegionSchedulerSuite{})
+
+type testBalanceHotRegionSchedulerSuite struct{}
+
+func (s *testBalanceHotRegionSchedulerSuite) TestBalance(c *C) {
+	cluster := newClusterInfo(newMockIDAllocator())
+	tc := newTestClusterInfo(cluster)
+
+	_, opt := newTestScheduleConfig()
+	hb := newBalanceHotRegionScheduler(opt)
+	// Add stores 1,2,3,4
+	tc.addRegionStore(1, 3)
+	tc.addRegionStore(2, 3)
+	tc.addRegionStore(3, 3)
+	tc.addRegionStore(4, 0)
+	tc.updateStorageWrittenBytes(1, 165062213)
+	tc.updateStorageWrittenBytes(2, 0)
+	tc.updateStorageWrittenBytes(3, 0)
+
+	// Add 3 regions
+	//| region_id | leader_sotre | follower_store | follower_store | written_bytes |
+	//|-----------|--------------|----------------|----------------|---------------|
+	//|     1     |       1      |        2       |       3        |     479516    |
+	//|     2     |       1      |        2       |       -        |     418115    |
+	//|     3     |       1      |        2       |       3        |       0       |
+
+	tc.addLeaderRegionWithWriteInfo(1, 1, 479516*regionHeartBeatReportInterval, 2, 3)
+	// for test follow just 1 peer here
+	tc.addLeaderRegionWithWriteInfo(2, 1, 419115*regionHeartBeatReportInterval, 2)
+	tc.addLeaderRegionWithWriteInfo(3, 1, 0, 2, 3)
+	hotRegionLowThreshold = 0
+	checkTransferLeader(c, hb.Schedule(cluster), 1, 2)
```

Comment:
Is region2 that is be balanced? Why not balance region1 which has a greater written_bytes?